packages/payload/src/express/middleware/errorHandler.ts

```diff
@@ -51,7 +51,7 @@ const errorHandler =
       )) || { response, status })
     }
 
-    if (typeof config.hooks.afterError === 'function') {
+    if (req.collection && typeof config.hooks.afterError === 'function') {
```

Comment:
This shouldn't be changed here because we still want to call the `afterError` in the payload config.

Instead we should optionally chain `req.collection.config` on line 59 to safely handle this. Then we need to also change the type for the `afterError` function 4th arg to be optional.

Otherwise if we just take your change there is the chance that projects have custom `afterError` hooks that won't be called.